### PR TITLE
cf-guest: implement light client for the guest blockchain

### PR DESCRIPTION
**Pull request by mina86**
Add partial IBC light client implementation for the guest blockchain.
Misbehaviour handling isn’t yet done.

### CODE REVIEW

File: common/ibc-guest/proto/guest.proto [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	// Borsh-serialised epoch the block belongs to.  The epoch specifies
+	// validators which sign the block.
+	bytes epoch = 3;
+
+	repeated Signature signatures = 4;
+}
+
+message Signature {
+	uint64 index = 1;
```
Comment (blasrodri) [inline common/ibc-guest/proto/guest.proto:49]: what's this index?

File: common/ibc-guest/src/client.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    pub fn with_header(&self, header: &super::Header<PK>) -> Self {
+        let latest_height = core::cmp::max(
+            header.block_header.block_height,
+            self.latest_height,
+        );
+        let epoch_commitment = header
+            .block_header
+            .next_epoch_commitment
+            .as_ref()
+            .unwrap_or(&self.epoch_commitment)
```
Comment (blasrodri) [inline common/ibc-guest/src/client.rs:42]: can you comment when it's ok not to use next_epoch_commitment?

File: common/ibc-guest/proto/guest.proto (hunk 1 above)
Comment (mina86) [reply common/ibc-guest/proto/guest.proto]: That’s the index in validator set defined in Epoch.

File: common/cf-guest/build.rs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+fn main() -> std::io::Result<()> {
+    prost_build::Config::new()
+        .enable_type_names()
+        .type_name_domain(["."], "composable.finance")
```
Comment (blasrodri) [inline common/cf-guest/build.rs:4]: can we change to Picasso?

Comment (mina86) [reply common/cf-guest/build.rs:4]: The idea is that the domain corresponds to an actual web domain. Though I guess sit wouldn’t be end of the world if we changed it to picasso.

**Pull request merged**